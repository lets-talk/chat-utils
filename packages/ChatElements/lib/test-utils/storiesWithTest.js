import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests(jestTestResults, {
  filesExt: '__tests__/**.js',
});
