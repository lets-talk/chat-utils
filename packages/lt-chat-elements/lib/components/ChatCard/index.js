import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Flex, Box } from 'grid-styled';

import { ellipsis } from '../../utils/style';

const ChatCard = (props) => {
  const {
    title, body, showRenderProp, render,
  } = props;

  const StyledCardTitle = styled.div`
    color: ${(innerProps) => innerProps.theme.palette.common.black};
    font-size: ${(innerProps) => innerProps.theme.typography.classes.caption.normal.dark.primary.fontSize};
    font-weight: ${(innerProps) => innerProps.theme.typography.weights.fontWeightBold};
    display: inline-block;
    vertical-align: middle;
    line-height: 1.2;
    padding-top: 2px;
    ${ellipsis('100%')}
  `;

  const StyledCardBody = styled.div`
    margin: 5px 0 0;
    font-size: 14px;
    overflow: hidden;
    line-height: 22px;
    color: #6e7a89;
    -webkit-animation: intercom-notification-body-animation .5s ease;
    animation: intercom-notification-body-animation .5s ease;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
  `;

  const StyledCardContainer = styled.div`
    max-width: ${props.width}px;
    max-height: ${props.height}px;
    padding: 15px;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    background-color: #fff;
    border-radius: 6px;
    box-shadow: ${(innerProps) => innerProps.theme.shadows['8p']};
    font-size: 13px;
    line-height: 1.5;
    height: 100%;
    position: relative;
  `;

  return (
    <StyledCardContainer>
      <Flex flex="1 1 auto" alignItems="center" pl={2}>
        {showRenderProp && render()}
        <StyledCardTitle>{title}</StyledCardTitle>
      </Flex>
      <Flex flex="1 1 auto">
        <Box p={2} ml="auto" width={1}>
          <StyledCardBody>{body}</StyledCardBody>
        </Box>
      </Flex>
    </StyledCardContainer>
  );
};


ChatCard.propTypes = {
  /**
   * types: default, rounded, circle
   */
  title: PropTypes.string,
  /**
   * types: default, rounded, circle
   */
  body: PropTypes.string,
  /**
   * width: Set the width of the notification
   */
  width: PropTypes.number,
  /**
   * height: Set the height of the card
   */
  height: PropTypes.number,
  /**
   * showRenderProp: Specify if render prop will be called
   */
  showRenderProp: PropTypes.bool,
  /**
   * render: function with a render prop to be show on the title left corner
   */
  render: PropTypes.func,
};

ChatCard.defaultProps = {
  title: '',
  body: '',
  height: 150,
  width: 250,
  showRenderProp: false,
};

export default ChatCard;
