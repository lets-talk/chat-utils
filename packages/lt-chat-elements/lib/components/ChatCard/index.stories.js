import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { withKnobs, text, number, select, boolean } from '@storybook/addon-knobs';

import { WrapWithTheme, defaultTheme, themeOptions } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';
// Component to show on storybook
import ChatCard from './';

const stories = storiesOf('ChatCard', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('ChatCard'));

stories.addWithInfo(
  'ChatCard',
  'This is the basic usage with the ChatCard with providing a label to show the text.',
  () => (
    <div
      style={{
        position: 'absolute', top: '50%', left: '50%', marginTop: '50px', marginLeft: '50px',
      }}
    >
      <WrapWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
      >
        <ChatCard
          title={text('Title', 'Hey There!!')}
          body={text('Content', 'Questions? We can help!')}
          width={number('Width', 200)}
          height={number('Height', 100)}
          margin={text('Margin', '0px 10px 0px 0px')}
          showRenderProp={boolean('showRenderProp', false)}
          render={() => <h3>Can show any Component </h3>}
        />
      </WrapWithTheme>
    </div>
  )
);
