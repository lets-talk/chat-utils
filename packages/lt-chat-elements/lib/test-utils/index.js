// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
