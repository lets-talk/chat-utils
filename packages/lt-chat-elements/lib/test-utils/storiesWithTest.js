import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
