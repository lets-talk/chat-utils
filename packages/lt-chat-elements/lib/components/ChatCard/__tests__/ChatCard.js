import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import ChatCard from '../index';
import { mountWithTheme, renderWithTheme } from '../../../test-utils';

describe('ChatCard component', () => {
  it('should render with no props', () => {
    const component = shallow(<ChatCard />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const props = {
      title: 'Example Card',
      body: 'This is an example card content',
      height: 50,
      width: 50,
      showRenderProp: true,
      render: () => <h1>Render child</h1>,
    };

    const component = shallow(<ChatCard {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();

    expect(component.find('StyledCardTitle').children().text()).toContain('Example Card');
    expect(component.find('StyledCardBody').children().text()).toContain('This is an example card content');
  });

  it('should render with custom props', () => {
    const props = {
      title: 'Example Card',
      body: 'This is an example card content',
      height: 50,
      width: 150,
      showRenderProp: false,
    };

    const component = mountWithTheme(<ChatCard {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();

    expect(component.find('StyledCardTitle').children().text()).toContain('Example Card');
    expect(component.find('StyledCardBody').children().text()).toContain('This is an example card content');
  });

  it('Styled: should have the correct styles (Those that depends on the theme)', () => {
    const props = {
      title: 'Example Card',
      body: 'This is an example card content',
      height: 50,
      width: 150,
      showRenderProp: false,
    };
    const tree = renderWithTheme(<ChatCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
