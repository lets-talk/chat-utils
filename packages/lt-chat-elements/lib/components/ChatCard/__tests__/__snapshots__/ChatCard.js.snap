// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatCard component Styled: should have the correct styles (Those that depends on the theme) 1`] = `
.c4 {
  box-sizing: border-box;
  width: 100%;
  margin-left: auto;
  padding: 8px;
}

.c1 {
  box-sizing: border-box;
  padding-left: 8px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  box-sizing: border-box;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  max-width: 150px;
  max-height: 50px;
  padding: 15px;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  background-color: rgb(255,255,255);
  border-radius: 6px;
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.14),0 3px 14px 2px rgba(0,0,0,0.12),0 5px 5px -3px rgba(0,0,0,0.2);
  font-size: calc(12 * 1px);
  line-height: calc(16 * 1px);
  height: 100%;
  position: relative;
}

.c5 {
  margin: 5px 0 0;
  font-size: 14px;
  overflow: hidden;
  line-height: 22px;
  color: #6e7a89;
  -webkit-animation: intercom-notification-body-animation .5s ease;
  -webkit-animation: intercom-notification-body-animation .5s ease;
  animation: intercom-notification-body-animation .5s ease;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

.c2 {
  color: rgba(0,0,0,0.87);
  font-size: calc(12 * 1px);
  font-weight: 700;
  display: inline-block;
  vertical-align: middle;
  line-height: 1.2;
  padding-top: 2px;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Example Card
    </div>
  </div>
  <div
    className="c3"
  >
    <div
      className="c4"
    >
      <div
        className="c5"
      >
        This is an example card content
      </div>
    </div>
  </div>
</div>
`;

exports[`ChatCard component should render with custom props 1`] = `
<StyledCardContainer>
  <Flex
    alignItems="center"
    flex="1 1 auto"
    pl={2}
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <h1>
      Render child
    </h1>
    <StyledCardTitle>
      Example Card
    </StyledCardTitle>
  </Flex>
  <Flex
    flex="1 1 auto"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      ml="auto"
      p={2}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <StyledCardBody>
        This is an example card content
      </StyledCardBody>
    </Box>
  </Flex>
</StyledCardContainer>
`;

exports[`ChatCard component should render with custom props 2`] = `
.c4 {
  box-sizing: border-box;
  width: 100%;
  margin-left: auto;
  padding: 8px;
}

.c1 {
  box-sizing: border-box;
  padding-left: 8px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  box-sizing: border-box;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  max-width: 150px;
  max-height: 50px;
  padding: 15px;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  background-color: rgb(255,255,255);
  border-radius: 6px;
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.14),0 3px 14px 2px rgba(0,0,0,0.12),0 5px 5px -3px rgba(0,0,0,0.2);
  font-size: calc(12 * 1px);
  line-height: calc(16 * 1px);
  height: 100%;
  position: relative;
}

.c5 {
  margin: 5px 0 0;
  font-size: 14px;
  overflow: hidden;
  line-height: 22px;
  color: #6e7a89;
  -webkit-animation: intercom-notification-body-animation .5s ease;
  -webkit-animation: intercom-notification-body-animation .5s ease;
  animation: intercom-notification-body-animation .5s ease;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

.c2 {
  color: rgba(0,0,0,0.87);
  font-size: calc(12 * 1px);
  font-weight: 700;
  display: inline-block;
  vertical-align: middle;
  line-height: 1.2;
  padding-top: 2px;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<ChatCard
  body="This is an example card content"
  height={50}
  showRenderProp={false}
  title="Example Card"
  width={150}
>
  <StyledCardContainer>
    <div
      className="c0"
    >
      <Flex
        alignItems="center"
        flex="1 1 auto"
        pl={2}
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
      >
        <Clean.div
          alignItems="center"
          blacklist={
            Array [
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
              "w",
              "fontSize",
              "f",
              "color",
              "bg",
              "fontFamily",
              "font",
              "textAlign",
              "align",
              "lineHeight",
              "fontWeight",
              "letterSpacing",
              "display",
              "maxWidth",
              "minWidth",
              "height",
              "maxHeight",
              "minHeight",
              "size",
              "ratio",
              "alignItems",
              "alignContent",
              "justifyContent",
              "justify",
              "flexWrap",
              "wrap",
              "flexBasis",
              "flexDirection",
              "flex",
              "justifySelf",
              "alignSelf",
              "order",
              "gridGap",
              "gridColumnGap",
              "gridRowGap",
              "gridColumn",
              "gridRow",
              "gridAutoFlow",
              "gridAutoColumns",
              "gridAutoRows",
              "gridTemplateColumns",
              "gridTemplateRows",
              "border",
              "borderTop",
              "borderRight",
              "borderBottom",
              "borderLeft",
              "borderColor",
              "borderRadius",
              "boxShadow",
              "background",
              "backgroundImage",
              "bgImage",
              "backgroundSize",
              "bgSize",
              "backgroundPosition",
              "bgPosition",
              "backgroundRepeat",
              "bgRepeat",
              "position",
              "zIndex",
              "top",
              "right",
              "bottom",
              "left",
              "hover",
              "focus",
              "active",
              "disabledStyle",
              "textStyle",
              "colors",
              "buttonStyle",
              "borderWidth",
            ]
          }
          className="c1"
          flex="1 1 auto"
          is="div"
          pl={2}
          theme={
            Object {
              "breakpoints": Array [
                "40em",
                "52em",
                "64em",
              ],
              "fontSizes": Array [
                12,
                14,
                16,
                20,
                24,
                32,
                48,
                64,
                96,
                128,
              ],
              "space": Array [
                0,
                4,
                8,
                16,
                32,
                64,
                128,
                256,
                512,
              ],
            }
          }
        >
          <div
            className="c1"
          >
            <StyledCardTitle>
              <div
                className="c2"
              >
                Example Card
              </div>
            </StyledCardTitle>
          </div>
        </Clean.div>
      </Flex>
      <Flex
        flex="1 1 auto"
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
      >
        <Clean.div
          blacklist={
            Array [
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
              "w",
              "fontSize",
              "f",
              "color",
              "bg",
              "fontFamily",
              "font",
              "textAlign",
              "align",
              "lineHeight",
              "fontWeight",
              "letterSpacing",
              "display",
              "maxWidth",
              "minWidth",
              "height",
              "maxHeight",
              "minHeight",
              "size",
              "ratio",
              "alignItems",
              "alignContent",
              "justifyContent",
              "justify",
              "flexWrap",
              "wrap",
              "flexBasis",
              "flexDirection",
              "flex",
              "justifySelf",
              "alignSelf",
              "order",
              "gridGap",
              "gridColumnGap",
              "gridRowGap",
              "gridColumn",
              "gridRow",
              "gridAutoFlow",
              "gridAutoColumns",
              "gridAutoRows",
              "gridTemplateColumns",
              "gridTemplateRows",
              "border",
              "borderTop",
              "borderRight",
              "borderBottom",
              "borderLeft",
              "borderColor",
              "borderRadius",
              "boxShadow",
              "background",
              "backgroundImage",
              "bgImage",
              "backgroundSize",
              "bgSize",
              "backgroundPosition",
              "bgPosition",
              "backgroundRepeat",
              "bgRepeat",
              "position",
              "zIndex",
              "top",
              "right",
              "bottom",
              "left",
              "hover",
              "focus",
              "active",
              "disabledStyle",
              "textStyle",
              "colors",
              "buttonStyle",
              "borderWidth",
            ]
          }
          className="c3"
          flex="1 1 auto"
          is="div"
          theme={
            Object {
              "breakpoints": Array [
                "40em",
                "52em",
                "64em",
              ],
              "fontSizes": Array [
                12,
                14,
                16,
                20,
                24,
                32,
                48,
                64,
                96,
                128,
              ],
              "space": Array [
                0,
                4,
                8,
                16,
                32,
                64,
                128,
                256,
                512,
              ],
            }
          }
        >
          <div
            className="c3"
          >
            <Box
              ml="auto"
              p={2}
              theme={
                Object {
                  "breakpoints": Array [
                    "40em",
                    "52em",
                    "64em",
                  ],
                  "fontSizes": Array [
                    12,
                    14,
                    16,
                    20,
                    24,
                    32,
                    48,
                    64,
                    96,
                    128,
                  ],
                  "space": Array [
                    0,
                    4,
                    8,
                    16,
                    32,
                    64,
                    128,
                    256,
                    512,
                  ],
                }
              }
              width={1}
            >
              <Clean.div
                blacklist={
                  Array [
                    "m",
                    "mt",
                    "mr",
                    "mb",
                    "ml",
                    "mx",
                    "my",
                    "p",
                    "pt",
                    "pr",
                    "pb",
                    "pl",
                    "px",
                    "py",
                    "width",
                    "w",
                    "fontSize",
                    "f",
                    "color",
                    "bg",
                    "fontFamily",
                    "font",
                    "textAlign",
                    "align",
                    "lineHeight",
                    "fontWeight",
                    "letterSpacing",
                    "display",
                    "maxWidth",
                    "minWidth",
                    "height",
                    "maxHeight",
                    "minHeight",
                    "size",
                    "ratio",
                    "alignItems",
                    "alignContent",
                    "justifyContent",
                    "justify",
                    "flexWrap",
                    "wrap",
                    "flexBasis",
                    "flexDirection",
                    "flex",
                    "justifySelf",
                    "alignSelf",
                    "order",
                    "gridGap",
                    "gridColumnGap",
                    "gridRowGap",
                    "gridColumn",
                    "gridRow",
                    "gridAutoFlow",
                    "gridAutoColumns",
                    "gridAutoRows",
                    "gridTemplateColumns",
                    "gridTemplateRows",
                    "border",
                    "borderTop",
                    "borderRight",
                    "borderBottom",
                    "borderLeft",
                    "borderColor",
                    "borderRadius",
                    "boxShadow",
                    "background",
                    "backgroundImage",
                    "bgImage",
                    "backgroundSize",
                    "bgSize",
                    "backgroundPosition",
                    "bgPosition",
                    "backgroundRepeat",
                    "bgRepeat",
                    "position",
                    "zIndex",
                    "top",
                    "right",
                    "bottom",
                    "left",
                    "hover",
                    "focus",
                    "active",
                    "disabledStyle",
                    "textStyle",
                    "colors",
                    "buttonStyle",
                    "borderWidth",
                  ]
                }
                className="c4"
                is="div"
                ml="auto"
                p={2}
                theme={
                  Object {
                    "breakpoints": Array [
                      "40em",
                      "52em",
                      "64em",
                    ],
                    "fontSizes": Array [
                      12,
                      14,
                      16,
                      20,
                      24,
                      32,
                      48,
                      64,
                      96,
                      128,
                    ],
                    "space": Array [
                      0,
                      4,
                      8,
                      16,
                      32,
                      64,
                      128,
                      256,
                      512,
                    ],
                  }
                }
                width={1}
              >
                <div
                  className="c4"
                >
                  <StyledCardBody>
                    <div
                      className="c5"
                    >
                      This is an example card content
                    </div>
                  </StyledCardBody>
                </div>
              </Clean.div>
            </Box>
          </div>
        </Clean.div>
      </Flex>
    </div>
  </StyledCardContainer>
</ChatCard>
`;

exports[`ChatCard component should render with no props 1`] = `
<StyledCardContainer>
  <Flex
    alignItems="center"
    flex="1 1 auto"
    pl={2}
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <StyledCardTitle />
  </Flex>
  <Flex
    flex="1 1 auto"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      ml="auto"
      p={2}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <StyledCardBody />
    </Box>
  </Flex>
</StyledCardContainer>
`;
